tests/testthat/test-corridor.R

```diff
@@ -1,11 +1,26 @@
-test_that("river_buffer parameters can be configured via initial_corridor", {
+test_that("proper parameters must be provided depending on selected method", {
   river <- bucharest_osm$river_centerline
-  actual <- initial_corridor(river, buffer = 1)
-  expected <- river_buffer(river, buffer = 1)
-  expect_setequal(actual, expected)
+
+  # for "buffer" method, we need the "buffer" parameter
+  expect_error(initial_corridor(river, method = "buffer"),
+               "Buffer should be provided if `method` is `'buffer'`")
+  with_mocked_bindings(river_buffer = function(...) NULL, {
+    expect_no_error(initial_corridor(river, method = "buffer", buffer = 42))
+  })
```

Comment:
Okay, so we mock the `river_buffer()` function to only check if both arguments are passed to it.

Reply:
Exactly! More importantly than for the river buffer, I want to run the same test for the river valley, which is much more expensive (and require the actual DEM to be available)